src/com/shmuelzon/HomeAssistantFloorPlan/Controller.java

```diff
@@ -962,14 +962,30 @@ private Point2d getCenterOfStateIcons(Set<Entity> entities) {
     private void separateStateIcons(Set<Entity> entities) {
         final double STEP_SIZE = 2.0;
 
-        Point2d centerPostition = getCenterOfStateIcons(entities);
+        if (entities.isEmpty()) {
+            return;
+        }
+
+        Entity entityToMove = entities.iterator().next();
```

Comment:
Could this always return the same entity in the intersecting group? So if, for example, we have 2 that intersect, one will stay in it's place while the other would move?

The original implementation had the all move further away from the center so they would all revolve around whatever furniture they represent and I'd like to keep it that way.

Reply:
Fair point. I guess the center of the remaining overlapping entities does shift a bit in relation to all entities.

Using the loop we had before + setting a direction when the vector length == 0, will not work. Entities with the same position will al shift the same.

I'm not sure yet about solving this. Maybe storing the original center-position as a private class property so it can be referred to in different calls of method `separateStateIcons()`?

Reply:
The problem is that with each iteration of looking for overlapping icons, you might create new sets where icons that didn't overlap before, might do now after things were moved around. I think we'd need to define some default direction when the direction is zero but apply it with different signs (negative/positive) to each entity. However, I'd want the sign to always be the same for each entity and not random so that each time the plugin is used, the results would be the same. Still have no idea how to do that nicely....

Reply:
I've created another way to distribute the icons in a nice way.
E.g. Move the icon away in the calculated direction if it has one (current method), or else distribute the icons around the center-position.

![image](https://github.com/user-attachments/assets/12e29681-5812-4581-a626-a9cfc0df6a8c)

The thing that bothers me...
Once an icon has moved that far that it won't overlap anymore, the next call of `findIntersectingStateIcons()` & `separateStateIcons()`, changes the set of overlapping icons , and therefor the center-position in relation to the original set.

I think we actually want to keep moving icons away from the center-position of the original set of overlapping icons into the direction defined as first.

Reply:
But what do we do if those then start colliding with another group? If we don't change the center then we'll just increase the radius until they just happen to not overlap but by then they might be so far away from the original center that it's not a relevant position anymore